packages/projects/default-project/PhysicsSimulationTestSystem.ts

```diff
@@ -0,0 +1,171 @@
+import { createNewEditorNode } from '@xrengine/engine/src/scene/functions/SceneLoading'
+import { TransformComponent } from '@xrengine/engine/src/transform/components/TransformComponent'
+
+// Maybe do this using system injection node
+// receiveActionOnce(EngineEvents.EVENTS.SCENE_LOADED, () => {
+//     console.log("gltf parse scene loaded")
+//     generateSimulationData(5)
+// })
+
+/**
+ * Returns a random number between min (inclusive) and max (exclusive)
+ */
+function getRandomArbitrary(min, max) {
+  return Math.random() * (max - min) + min
+}
+
+/**
+ * Returns a random integer between min (inclusive) and max (inclusive).
+ * The value is no lower than min (or the next integer greater than min
+ * if min isn't an integer) and no greater than max (or the next integer
+ * lower than max if max isn't an integer).
+ * Using Math.round() will give you a non-uniform distribution!
+ */
+function getRandomInt(min, max) {
+  min = Math.ceil(min)
+  max = Math.floor(max)
+  return Math.floor(Math.random() * (max - min + 1)) + min
+}
+
+let uuidCounter = 0
+function getUUID() {
+  uuidCounter++
+  return 'physicstestuuid' + uuidCounter
+}
+
+let simulationObjectsGenerated = false
+export default async function PhysicsSimulationTestSystem(world: World) {
+  return () => {
+    if (!Engine.isInitialized || !world.physics.physics || simulationObjectsGenerated) return
+    simulationObjectsGenerated = true
+    generateSimulationData(0)
+  }
+}
+
+export const generateSimulationData = (numOfObjectsToGenerate) => {
+  // Auto generate objects
+  for (let index = 0; index < numOfObjectsToGenerate; index++) {
+    let config = {} as ShapeOptions
+    config.type = getRandomInt(0, 1) ? 'box' : 'sphere'
+    config.bodyType = getRandomInt(0, 2)
+    config.collisionLayer = CollisionGroups.Default
+    config.collisionMask = CollisionGroups.Default | CollisionGroups.Avatars
+    config.staticFriction = 1
+    config.dynamicFriction = 1
+    config.restitution = 1
+
+    console.log('generating object with config:', config)
+    generatePhysicsObject(config)
+  }
+
+  // Define and generate any objects with manual configs
+  let config = {
+    type: 'sphere' as ColliderTypes,
+    bodyType: BodyType.DYNAMIC,
+    collisionLayer: CollisionGroups.Default,
+    collisionMask: CollisionGroups.Default | CollisionGroups.Avatars,
+    staticFriction: 1,
+    dynamicFriction: 1,
+    restitution: 0.1
+  }
+
+  // TODO: Generating the object as a network object here may result in an error sometimes,
+  // the reason being the spawn object network action from server is received before the object has been added to the scene locally.
+  // In order to test network functionality, call the generateSimulationData directly from scene loading logic in code for now.
+  generatePhysicsObject(config, new Vector3(0, 15, 0))
+}
+
+const defaultSpawnPosition = new Vector3()
+export const generatePhysicsObject = (
```

Comment:
this is all great, we should pull bits of this out and use it in unit tests for the physics system

Reply:
Will do in the tests.